### ISSUE TITLE
linkerd inject breaks HPA configuration as no requests are set for proxy sidecar

### ISSUE THREAD

**Issue by benjdlambert** [linkerd/linkerd2#1480]
👋

After moving some of our services that use `HPA` to scale `Deployments` in `k8s`, the `HPA` will not be able to work out the current `cpu` usage as there is no default `requests` in the deployment for the `linkerd-proxy` sidecar that it creates.

```
Conditions:
  Type            Status  Reason                   Message
  ----            ------  ------                   -------
  AbleToScale     True    SucceededGetScale        the HPA controller was able to get the target's current scale
  ScalingActive   False   FailedGetResourceMetric  the HPA was unable to compute the replica count: missing request for cpu on container linkerd-proxy in pod ms/gql-7d485bd799-dnf66
  ScalingLimited  True    TooFewReplicas           the desired replica count is increasing faster than the maximum scale rate
Events:
  Type     Reason                   Age                From                       Message
  ----     ------                   ----               ----                       -------
  Warning  FailedGetResourceMetric  4m (x253 over 2h)  horizontal-pod-autoscaler  missing request for cpu on container linkerd-proxy in pod ms/gql-7d485bd799-dnf66
```

Maybe we can add some sensible defaults into the `linkerd inject`?

**Comment by grampelberg** [comment]
@benjdlambert I *believe* you mentioned that adding the resource requests still freaked HPA out. Did it end up just working in the end?

**Comment by benjdlambert** [comment]
@grampelberg - so this is an interesting one. We added some resource requests, that were /kindof/ sane. `100m CPU`+ `50Mi mem` but bascially HPA just decided to scale it to the `maxReplicas` because it couldn't get it to settle to our `targetCPU` which was set at 30%. I need to investigate more, and play around maybe with the latest `HPA` to see if it's a problem with our configuration (most likely), or if we're not giving enough resources to the sidecar. 

Maybe we keep this issue open, and I can give some bits a try over the weekend and report back?

/blam

**Comment by grampelberg** [comment]
Please do, I'm going to be begging for a doc on using HPA with linkerd immediately after that =)

**Comment by benjdlambert** [comment]
OK - so @grampelberg I think I've fixed a nice solution.

I think what we were seeing was `heapster` and `metrics-server` fighting, and it was reporting some weird metrics causing everything to scale out.

We've changed our target CPU to 70%, giving `cpu: 100m` and `mem: 50Mi` seems to work OK right now at least.

I will monitor throughout the few days to see if it's working as expected, and will report back!
